lib/client/ipam_block_reader_writer.go

```diff
@@ -139,6 +124,20 @@ func (rw blockReaderWriter) claimNewAffineBlock(
 	return nil, noFreeBlocksError("No Free Blocks")
 }
 
+// isIPPoolInRequestedPools checks if the IP Pool that is passed in belongs to the list of IP Pools
+// that should be used for assigning IPs from.
+func isIPPoolInRequestedPools(pool net.IPNet, requestedPools []cnet.IPNet) bool {
```

Comment:
This is a weird interface since you are comparing two different types.  It would make sense for both of these parameters to be cnet.IPNet - which would have simply required you to not deref the `CIDR.IPNet` above.  Then I think this method could be `isIPNetInIPNets` or similar since it is no longer strictly related to pools.